sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/statsEstimation/FilterEstimationSuite.scala

```diff
@@ -158,8 +196,8 @@ class FilterEstimationSuite extends StatsEstimationTestBase {
     val condition = Not(And(LessThan(attrInt, Literal(3)), Literal(null, IntegerType)))
     validateEstimatedStats(
       Filter(condition, childStatsTestPlan(Seq(attrInt), 10L)),
-      Seq(attrInt -> colStatInt.copy(distinctCount = 8)),
-      expectedRowCount = 8)
+      Seq(attrInt -> colStatInt.copy(distinctCount = 7)),
```

Comment:
Shall we add new test cases for filter estimation based on histogram, instead of modifying existing test results?

Reply:
+1